src/extension/infer.rs

```diff
@@ -109,51 +110,67 @@ pub enum InferExtensionError {
 
 /// A graph of metavariables which we've found equality constraints for. Edges
```

Comment:
Remove `equality` here. If this was only for equality constraints, there'd be no need for a graph - equality being reflexive and transitive, you'd just be dividing things into sets, right. It's only for Plus constraints that we need a graph?

Reply:
For the equality case we are dividing things into sets, but the mechanism for doing so is by finding the connected components of the undirected version of this graph